Can't access the tokens of header by JavaScript

I would like to access the tokens of header by JavaScript. But I can't do it.

## sample file

```test.md
# レベル１

## レベル2

```


## test 1

### code

```javascript
function validateSection(section) {
    console = {
        log:print,
    };
    console.log(section.headerContent[0])
}
```

### result

```java
[2016-05-09 22:55:26.359][ERROR] cc.redpen.validator.JavaScriptValidator - failed to invoke validateSection
javax.script.ScriptException: TypeError: Cannot read property "0" from undefined in <eval> at line number 7
        at jdk.nashorn.api.scripting.NashornScriptEngine.throwAsScriptException(NashornScriptEngine.java:467) ~[nashorn.jar:na]
        at jdk.nashorn.api.scripting.NashornScriptEngine.invokeImpl(NashornScriptEngine.java:389) ~[nashorn.jar:na]
        at jdk.nashorn.api.scripting.NashornScriptEngine.invokeFunction(NashornScriptEngine.java:190) ~[nashorn.jar:na]
        at cc.redpen.validator.JavaScriptValidator.call(JavaScriptValidator.java:159) [redpen-core-1.5.2.jar:na]
        at cc.redpen.validator.JavaScriptValidator.validate(JavaScriptValidator.java:144) [redpen-core-1.5.2.jar:na]
        at cc.redpen.RedPen.lambda$runSectionValidators$26(RedPen.java:188) [redpen-core-1.5.2.jar:na]
        at java.util.ArrayList.forEach(Unknown Source) ~[na:1.8.0_91]
        at cc.redpen.RedPen.runSectionValidators(RedPen.java:188) [redpen-core-1.5.2.jar:na]
        at cc.redpen.RedPen.validate(RedPen.java:134) [redpen-core-1.5.2.jar:na]
        at cc.redpen.Main.run(Main.java:175) ~[redpen-cli-1.5.2.jar:na]
        at cc.redpen.Main.main(Main.java:59) ~[redpen-cli-1.5.2.jar:na]
Caused by: jdk.nashorn.internal.runtime.ECMAException: TypeError: Cannot read property "0" from undefined
        at jdk.nashorn.internal.runtime.ECMAErrors.error(ECMAErrors.java:57) ~[nashorn.jar:na]
        at jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:213) ~[nashorn.jar:na]
        at jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:185) ~[nashorn.jar:na]
        at jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:172) ~[nashorn.jar:na]
        at jdk.nashorn.internal.runtime.Undefined.get(Undefined.java:157) ~[nashorn.jar:na]
        at jdk.nashorn.internal.scripts.Script$Recompilation$36$34A$\^eval\_.validateSection(<eval>:7) ~[na:na]
        at jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:625) ~[nashorn.jar:na]
        at jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:494) ~[nashorn.jar:na]
        at jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:393) ~[nashorn.jar:na]
        at jdk.nashorn.api.scripting.ScriptObjectMirror.callMember(ScriptObjectMirror.java:199) ~[nashorn.jar:na]
        at jdk.nashorn.api.scripting.NashornScriptEngine.invokeImpl(NashornScriptEngine.java:383) ~[nashorn.jar:na]
        ... 9 common frames omitted
```

## test 2

### code

```javascript
function validateSection(section) {
    console = {
        log:print,
    };

    console.log(section.getHeaderContents())
}
```

### result

 A part of result is garbled due to my powershell setting.

```java
[2016-05-09 23:05:12.698][INFO ] cc.redpen.validator.JavaScriptValidator - JavaScript validators directory: js
[Sentence{links=[], lineNumber=0, content='', startPositionOffset=0, isFirstSentence=false, tokens=[], offsetMap=[]}]
[Sentence{links=[], lineNumber=1, content='繝ｬ繝吶Ν・・, startPositionOffset=2, isFirstSentence=true, tokens=[], offsetMap=[LineOffset(1, 2), LineOffset(1, 3), LineOffset(1, 4), LineOffset(1, 5)]}]
[Sentence{links=[], lineNumber=3, content='繝ｬ繝吶Ν2', startPositionOffset=3, isFirstSentence=true, tokens=[], offsetMap=[LineOffset(3, 3), LineOffset(3, 4), LineOffset(3, 5), LineOffset(3, 6)]}]
```

--- a/redpen-core/src/main/java/cc/redpen/model/Document.java
+++ b/redpen-core/src/main/java/cc/redpen/model/Document.java
@@ -160,6 +160,9 @@ public class Document implements Iterable<Section>, Serializable {
          */
         public DocumentBuilder appendSection(Section section) {
             ensureNotBuilt();
+            for (Sentence sentence : section.getHeaderContents()) {
+                sentence.setTokens(tokenizer.tokenize(sentence.getContent()));
+            }
             sections.add(section);
             return this;
         }

--- a/redpen-core/src/test/java/cc/redpen/MarkdownParserTest.java
+++ b/redpen-core/src/test/java/cc/redpen/MarkdownParserTest.java
@@ -539,6 +539,21 @@ public class MarkdownParserTest extends BaseParserTest {
     }
 
     @Test
+    public void testAccessTokensInHeader() {
+        String sampleText = "# This is level one section\n";
+        sampleText += "## This is level two section\n";
+
+        Document doc = createFileContent(sampleText);
+        assertEquals(3, doc.size());
+        assertEquals(0, doc.getSection(0).getLevel());
+        assertEquals(1, doc.getSection(1).getLevel());
+        assertEquals(2, doc.getSection(2).getLevel());
+
+        assertEquals("This is level one section", doc.getSection(1).getHeaderContent(0).getContent());
+        assertEquals(5, doc.getSection(1).getHeaderContent(0).getTokens().size());
+    }
+
+    @Test
     public void testDocumentWithHeaderWithoutPeriods()
             throws UnsupportedEncodingException {
         String sampleText = "";

